src/Features/Core/Portable/EditAndContinue/EditAndContinueDiagnosticUpdateSource.cs

```diff
@@ -147,5 +147,38 @@ public void ReportDiagnostics(Workspace workspace, Solution solution, ImmutableA
                     diagnostics: solutionDiagnostics));
             }
         }
+
+        private static Dictionary<K, ImmutableArray<T>> ToDictionary<K, T>(
```

Comment:
dedicated IDE copy, since this is hte only place we use this helper.

Reply:
@tmat we could also just replace this with a GroupBy call above instead, which woudl likely be good enough for this diagnostic source (unless you expect *massive* amounts of diagnostics here).

Reply:
Feel free to replace with GroupBy.

Reply:
thanks!